### DISCUSSION TITLE
How to tell whether a library is static or shared?

### DISCUSSION THREAD

**Discussion by jwillikers** [mesonbuild/meson#12958] [Q&A] [answered]
When adding a library with the `library` function, how do you tell the type of that library? I need to only run an API-compatibility test for a library when it is built as a shared library.

**Comment by dcbaker** [comment, marked as answer]
`library()` is controlled by the `default_library` option. This can also include a "both" setting, so you probably want to test for "!static", something like:
```meson
lib = library(...)
if get_option('default_library') != 'static'
  if get_option('default_library') == 'both'
    lib = lib.get_shared_library()
  endif
  test(..., args : lib)
endif
```

**Reply by jwillikers** [reply]
Thanks, that's pretty close to what I'd reckoned, but I didn't specifically pull the shared library when both. I just had to change ` get_shared_library` to `get_shared_lib`.
